orion/pkg/domains/program.go

```diff
@@ -10,15 +9,21 @@ import (
 var TABLE_PROGRAMS = "programs"
 
 type Program struct {
-	Id          uint         `json:"id"`
-	CreatedAt   time.Time    `json:"-" db:"created_at"`
-	UpdatedAt   time.Time    `json:"-" db:"updated_at"`
-	DeletedAt   sql.NullTime `json:"-" db:"deleted_at"`
-	ProgramId   string       `json:"programId" db:"program_id"`
-	Name        string       `json:"name"`
-	Grade1      uint         `json:"grade1"`
-	Grade2      uint         `json:"grade2"`
-	Description string       `json:"description"`
+	Id          uint      `json:"id"`
+	CreatedAt   time.Time `json:"-" db:"created_at"`
+	UpdatedAt   time.Time `json:"-" db:"updated_at"`
+	DeletedAt   NullTime  `json:"-" db:"deleted_at"`
+	PublishedAt NullTime  `json:"publishedAt" db:"published_at"`
+	ProgramId   string    `json:"programId" db:"program_id"`
+	Name        string    `json:"name"`
+	Grade1      uint      `json:"grade1"`
+	Grade2      uint      `json:"grade2"`
+	Description string    `json:"description"`
+}
+
+type ProgramErrorBody struct {
+	ProgramId string `json:"programId"`
+	Error     error  `json:"error"`
```

Comment:
Regarding the above return response for the controllers.
You could use `c.JSON(status, <array of ProgramErrorBody>)`.